homeassistant/components/dreo/fan.py

```diff
@@ -0,0 +1,161 @@
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+from homeassistant.util.scaling import int_states_in_range
+
+from . import DreoConfigEntry
+from .entity import DreoEntity
+from .util import handle_api_exceptions
+
+FAN_SUFFIX = "fan"
+SCAN_INTERVAL = timedelta(seconds=10)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: DreoConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Fan from a config entry."""
+
+    async_add_entities(
+        [
+            DreoFan(device, config_entry)
+            for device in config_entry.runtime_data.devices
+            if DEVICE_TYPE.get(device.get("model")) == FAN_DEVICE.get("type")
+        ]
+    )
+
+
+class DreoFan(DreoEntity, FanEntity):
+    """Dreo fan."""
+
+    _attr_should_poll = True
+    _attr_supported_features = (
+        FanEntityFeature.PRESET_MODE
+        | FanEntityFeature.SET_SPEED
+        | FanEntityFeature.OSCILLATE
+        | FanEntityFeature.TURN_ON
+        | FanEntityFeature.TURN_OFF
+    )
+
+    def __init__(
+        self,
+        device: dict[str, Any],
+        config_entry: DreoConfigEntry,
+    ) -> None:
+        """Initialize the Dreo fan."""
+
+        super().__init__(device, config_entry, FAN_SUFFIX)
+        self._attr_name = None
+
+        model_config = FAN_DEVICE.get("config", {}).get(self._model, {})
+        speed_range = model_config.get("speed_range")
+
+        self._attr_preset_modes = model_config.get("preset_modes")
+        self._attr_preset_mode = None
+        self._attr_low_high_range = speed_range
+        self._attr_speed_count = int_states_in_range(speed_range)
+        self._attr_percentage = None
+        self._attr_oscillating = None
+        self._attr_is_on = False
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return device is state."""
+        return self._attr_is_on
+
+    @property
+    def preset_modes(self) -> list[str] | None:
+        """Return a list of available preset modes."""
+        return self._attr_preset_modes
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the preset mode of the fan."""
+        return self._attr_preset_mode
+
+    @property
+    def speed_count(self) -> int:
+        """Return the number of speeds the fan supports."""
+        return self._attr_speed_count
+
+    @property
+    def oscillating(self) -> bool | None:
+        """Return the oscillate of the fan."""
+        return self._attr_oscillating
+
+    def set_direction(self, direction: str) -> None:
+        """Set the direction of the fan."""
+        return
+
+    def turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn the device on."""
+        self._send_command("Turn the device on failed.", power_switch=True)
+        self._attr_is_on = True
```

Comment:
```suggestion
        self.schedule_update_ha_state(force_refresh=True)
```
instead of setting the attribute directly, can we get the state directly from the device?
`schedule_update_ha_state` with `force_refresh=True` calls the `update()` method, so it should fetch the new state from the device. or is the device slow to report the new state back?

Reply:
@abmantis Thanks for your suggestion! Based on our testing, our status retrieval response time is typically within 20 milliseconds, so it's completely feasible to use schedule_update_ha_state(force_refresh=True) after each control operation to get the latest device status.